source/frontend/src/features/properties/map/acquisition/add/AcquisitionProperties.tsx

```diff
@@ -32,11 +34,17 @@ export const AcquisitionProperties: React.FunctionComponent<
               <Col>
                 <MapSelectorContainer
                   addSelectedProperties={(newProperties: IMapProperty[]) => {
-                    newProperties.forEach((property, index) => {
+                    newProperties.forEach(async (property, index) => {
                       const formProperty = PropertyForm.fromMapProperty(property);
                       if (values.properties?.length === 0 && index === 0) {
                         formikProps.setFieldValue(`region`, formProperty.region);
                       }
+                      const address = property?.pid
+                        ? await getPrimaryAddressByPid(property.pid)
+                        : undefined;
+                      formProperty.address = address
+                        ? AddressForm.fromBcaAddress(address)
+                        : undefined;
```

Comment:
There is a subtle change in behaviour here by making the forEach callback async. There's the chance that the original order of the properties is not preserved in the case of an async promise to call BC Assessment say for property #2 takes way longer than property #10. 

In that case property-10 would be inserted in the formik array before property-2

If we really care for the order (I think we do) we should change this code to something that "serializes" the promises. Something like https://stackoverflow.com/questions/50263209/how-do-i-serialize-a-javascript-array-of-functions-returning-promises